core/opendaq/modulemanager/include/opendaq/module_manager.h

```diff
@@ -52,6 +58,24 @@ DECLARE_OPENDAQ_INTERFACE(IModuleManager, IBaseObject)
      * @param context The Context containing the Logger, Scheduler, Property Object Class Manager and Module Manager
      */
     virtual ErrCode INTERFACE_FUNC loadModules(IContext* context) = 0;
+
+    // [elementType(availableDevices, IDeviceInfo)]
+    /*!
+     * @brief Returns a list of known devices info.
+     * The implementation can start discovery in background and only return the results in this function.
+     * @param[out] availableDevices The list of known devices information.
+     */
+    virtual ErrCode INTERFACE_FUNC getAvailableDevices(IList** availableDevices) = 0;
+
+    /*!
+     * @brief Returns the device for specific connection string
+     * @param connectionString the connection string
+     * @param config the config
+     * @param parent the parent component
+     * @param logger the logger Component
+     * @param[out] device the device
+     */
+    virtual ErrCode INTERFACE_FUNC getDevice(IString* connectionString, IPropertyObject* config, IComponent* parent, ILoggerComponent* logger, IDevice** device) = 0;
```

Comment:
This should be `createDevice` or? And do we actually need that on the module manager?

Reply:
we ve discucced that we will move this thing from client to module manager